Add setState example to Basic Prop Types (#575)
typescript-cheatsheets/react: docs/basic/getting-started/basic-type-examples.md

  onClick(event: React.MouseEvent<HTMLButtonElement>): void;
  /** an optional prop (VERY COMMON!) */

>   optional?: OptionalType;
>   setState: React.Dispatch<React.SetStateAction<number>>; // the dispatch action returned by `useState`
> };

Reviewer: ```suggestion
  optional?: OptionalType;
  /** if you need to pass down the dispatch action returned by `useState` in a parent component, to a child component. we gave example of a "number" here but swap out with whatever type you have in useState */
  setState: React.Dispatch<React.SetStateAction<number>>;
};
```

  onClick(event: React.MouseEvent<HTMLButtonElement>): void;
  /** an optional prop (VERY COMMON!) */
  optional?: OptionalType;

>   /** if you need to pass down the dispatch action returned by `useState` in a parent component, to a child component. we gave example of a "number" here but swap out with whatever type you have in useState */

Reviewer 2: ```suggestion
  /** when passing down the state setter function to a child component. `number` is an example, swap out with whatever the type of your state */
```
